src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -140,21 +142,20 @@ class LibraryService (protected val argUserInfo: UserInfo,
     }
   }
 
+  // should only be used to change published state
   def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
     rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
       val pub = isPublished(workspaceResponse)
-      val requiredLevel = if (!pub || (pub && !value)) WorkspaceAccessLevels.Owner else WorkspaceAccessLevels.Write
-      if (workspaceResponse.accessLevel >= requiredLevel) {
-        val operations = updatePublishAttribute(value)
-        rawlsDAO.patchWorkspaceAttributes(ns, name, operations) map { ws =>
+      if (pub == value)
+        Future(RequestComplete(NoContent))
```

Comment:
if pub == value, but value is true, we still need to re-publish to ES, right? To ensure that the latest metadata gets to ES.